server.js

```diff
@@ -14,10 +17,72 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.post("/messages", (req, res) => {
+  if(req.body.from.length == 0){
+    res.status(400).json("Please enter a from property")
+    return
+  }
+  if(req.body.text.length == 0){
+    res.status(400).json("please enter a text property")
+    return
+  }
+  let message = {
+    id: messages.length,
+    from: req.body.from,
+    text: req.body.text,
+  };
+  messages.unshift(message);
+  res.json("message has been send successfully");
+});
+
+app.get("/messages/search", (req, res) =>{
+  console.log(req.query)
+  let newMessages = messages.filter((message)=>{
+    if(message.text.includes(req.query.text)){
```

Comment:
you should check the text querystring value is present first, and return a 400 if it is missing. Here it wont make a big difference, but in more complicated applications, it can provide a very large performance benefit.